# Урок 6. Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.

Задача 1

Даны значения величины заработной платы заемщиков банка (salary) и значения их поведенческого кредитного скоринга (scoring):

salary = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
scoring = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

1. Найдите ковариацию этих двух величин: сначала без использования специальных функций, а затем с помощью функции numpy.cov. Полученные значения должны быть равны.
2. Найдите коэффициент корреляции Пирсона: с помощью ковариации и средних квадратических отклонений двух признаков, а затем с помощью специальных функций из библиотек numpy и pandas.

In [11]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

salary = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
scoring = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [12]:
cov = np.mean(salary * scoring) - np.mean(salary) * np.mean(scoring)
cov

9157.839999999997

In [13]:
df = pd.DataFrame({'salary': salary, 'scoring': scoring})
np.cov(salary, scoring)[0, 1]

10175.377777777778

In [14]:
np.cov(salary, scoring, ddof=0)[0, 1]

9157.840000000002

In [15]:
matrix = np.corrcoef(salary, scoring)
matrix

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [18]:
from pydoc import help
from scipy.stats.stats import pearsonr
#help(pearsonr)

pearsonr(salary, scoring)

(0.8874900920739164, 0.0006107546587257491)

r>0,01≤0,29 – слабая положительная связь, 
r>0,30≤0,69 – умеренная положительная связь, 
r>0,70≤1,00 – сильная положительная связь, 
r>-0,01≤-0,29 – слабая отрицательная связь, 
r>-0,30≤-0,69 – умеренная отрицательная связь, 
r>-0,70≤-1,00 – сильная отрицательная связь.
https://kpfu.ru/portal/docs/F_2064674290/NPS_19.Pirson.Menshov.pdf

Задача 2

Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111

Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

http://meit.mgimo.ru/sites/default/files/CriticalValues.pdf  понадобилась таблица критических значений  z=1.96 критическое значение стандартного нормального распределения для уровня значимости α = 0,05

In [24]:
import math
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
iq_std = iq.std(ddof=1)
iq_std 
#стандартное отклонение

10.54566788359614

In [23]:
len(iq)

10

In [22]:
iq.mean()

118.1

In [25]:
iq.mean() - 1.96 * (iq_std / math.sqrt(len(iq)))

111.56372732786916

In [26]:
iq.mean() + 1.96 * (iq_std / math.sqrt(len(iq)))

124.63627267213083

доверительный интервал находится между 111.56372732786916 и 124.63627267213083

Задача 3

Известно, что рост футболистов в сборной распределен нормально с известной дисперсией 25. На выборке объёма 27 выборочное среднее составило 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [35]:
import random

D = 25
std = math.sqrt(D)
n = 27
x_average = 174.2
p = 1 - 0.95
a = x_average - D
b = x_average + D

In [36]:
x_average + 1.96 * (std / math.sqrt(n))

176.08601087935276

In [37]:
x_average - 1.96 * (std / math.sqrt(n))

172.31398912064722

доверительный интервал находится между 172.31398912064722 и 176.08601087935276

Задача 4

Выберите тему для проектной работы по курсу Теории вероятностей и математической статистики и напишите ее в комментарии к Практическому заданию.

Распределение по полу и возрасту по районам Калининградской области, данные за 3 года с Росстата. Если успею - добавлю обращения по поводу травм по территориям